src/theory/arith/nl/transcendental/sine_solver.cpp

```diff
@@ -244,13 +244,26 @@ void SineSolver::checkMonotonic()
                                d_data->d_zero,
                                d_data->d_pi_neg_2,
                                d_data->d_pi_neg};
-  std::vector<Node> mpoints_vals;
-
-  // get model values for points
-  for (const auto& point : mpoints)
+  // Sound lower (index=0), upper (index=1) bounds for the above points. We
+  // compute this by plugging in the upper and lower bound of pi.
+  std::vector<Node> mpointsBound[2];
+  TNode tpi = d_data->d_pi;
+  for (size_t j = 0; j < 5; j++)
   {
-    mpoints_vals.emplace_back(d_data->d_model.computeAbstractModelValue(point));
-    Assert(mpoints_vals.back().isConst());
+    Node point = mpoints[j];
+    for (size_t i = 0; i < 2; i++)
+    {
+      Node mpointapprox = point;
+      if (j != 2)
+      {
+        // bounds are flipped for negative pi
+        TNode tb = d_data->d_pi_bound[j > 2 ? 1 - i : i];
+        mpointapprox = point.substitute(tpi, tb);
+      }
+      mpointsBound[i].emplace_back(
+          d_data->d_model.computeAbstractModelValue(mpointapprox));
+      Assert(mpointsBound[i].back().isConst());
+    }
```

Comment:
I really don't like this loop... it seems super specific to `mpoints` and weirdly abstract at the same time.
My intuition is more along these lines:
```
for (const Node& point: mpoints)
{
  for (size_t i = 0; i < 2; ++i)
  {
    Node approx = point.substitute(d_data->d_pi_bound[i]);
    mpointsBound[i].emplace_back(d_data->d_model.computeConcreteModelValue(approx));
    Assert(mpointsBound[i].back().isConst());
  }
  if (mpointsBound[0].back().getConst<Rational>() > mpointsBound[1].back().getConst<Rational>())
  {
    std::swap(mpointsBound[0].back(), mpointsBound[1].back());
  }
}
```
This would be robust against changing the order of the evaluation points and is much more clear in why it sometimes flips the bounds (at least to me). Also note that I think we should use `computeConcreteModelValue()`, as they should do the very same thing here... right?